src/Component/PlaceOrder.js

```diff
@@ -0,0 +1,145 @@
+import React, { useContext } from "react";
+import "./PlaceOrder.css";
+import { useFormik } from "formik";
+import { CartContext } from "../App";
+import { useHistory } from "react-router-dom";
+import SignIn from "./SignIn";
+
+function PlaceOrder() {
+  const context = useContext(CartContext);
+  const history = useHistory();
+  const initialValues = {
+    name: "",
+    email: context.cartUserState ? context.cartUserState.email : "",
+    mob: "",
+    addr: "",
+    creditNo: "",
+  };
+  const onSubmit = (values) => {
+    alert("Order placed Successfully!");
+    context.dispatchState({
+      type: "clearCart",
+    });
+    history.push("/");
+  };
+  const validate = (values) => {
+    let errors = {};
+    if (!values.name) {
```

Comment:
what if we check falsy values against the pattern? I am trying to understand the reason for this branching.